server/pulp/server/webservices/views/search.py

```diff
@@ -154,6 +151,12 @@ def _generate_response(cls, query, options, *args, **kwargs):
         if hasattr(cls, 'model'):
             search_method = cls.model.objects.find_by_criteria
         else:
+            # id field should always be in query.fields. We are forced to do this twice, here and
+            # in the queryset for the mongoengine work. This is necessary because in mongoengine
+            # id is an alias to _id, and queries must be translated to reflect that. If we append
+            # id before this split, our serializers will translate this id also.
+            if query.fields and 'id' not in query.fields:
```

Comment:
@seandst, since this a `not in` scenario and 'id' is not in `[]`, it will attempt to run an append on `NoneType`. I can get the correct behavior hard coding `['id]` like I did above, but that is really just shorthand for:

``` python
if query.fields:
    if 'id' not in query.fields:
        query.fields.append('id')
else:
    if 'id' not in ['id']:
        query.fields.append('id')
```
Obviously, the last line will never run, but it still seems like a weird way to do it to me.